### PR TITLE
Workflow cleanups

### PR DESCRIPTION
**Pull request by mosuem**
Switch to using `pubspec_overrides.yaml` files instead of manually parsing and writing pubspec files. This makes local development easier while still testing the published version in CI. Implementing https://github.com/dart-lang/pub/issues/4076 would simplify this even further.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/native.yaml [hunk 1]
```diff
@@ -63,11 +63,8 @@ jobs:
           ndk-version: r26b
         if: ${{ matrix.sdk == 'stable' }}
 
-      - run: dart pub get -C ../../tools/
-        if: ${{ matrix.dependencies == 'path' }}
-
-      - run: dart ../../tools/bin/change_dependencies.dart 
-        if: ${{ matrix.dependencies == 'path' }}
+      - run: find -type f -name 'pubspec_overrides.yaml' -delete
```
Comment (dcharkes) [inline .github/workflows/native.yaml:66]: Does this work on Windows?

File: pkgs/native_assets_builder/pubspec_overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+dependency_overrides:
+  native_assets_cli:
+    path: ../native_assets_cli/
```
Comment (dcharkes) [inline pkgs/native_assets_builder/pubspec_overrides.yaml:3]: Here and other files, newline at the end

File: pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+dependency_overrides:
+  native_assets_cli:
+    path: ../../../../native_assets_cli/
```
Comment (dcharkes) [inline pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml:3]: How does this work when the package is copied to a temp folder? Ditto all the other test packages that get copied around to temp folders during tests.

**Review by dcharkes — COMMENTED** [review]
Can we split the PR in two, do the health thing in another PR? (Please add documentation to the health thing, I've never heard of it.)

I like the idea that the default developer experience is path dependencies. That makes it easier to make changes across multiple packages. I don't think that this will fully work with the tests though. When running the tests that copy testing packages to temp folders, the overrides inside those folders have paths that are no longer correct?

File: pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml (hunk 3 above)
Comment (mosuem) [reply pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml]: They now get the same handling as pubspec files.

**Comment by mosuem** [comment]
The CI was green as the overrides were not copied, so the packages were resolved via pub. Now the overrides are copied and point to the right package folders.

File: pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml (hunk 3 above)
Comment (dcharkes) [reply pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml]: Sweet!

So we just need to make sure that the transitive closure is in the override files right? Which I believe you've done correctly in this PR.

**Comment by dcharkes** [comment]
The PR description is out of date.

Please expand the description a bit with https://github.com/dart-lang/native/pull/846#discussion_r1415688162 so we can understand our own changes in 3 years. 😄

File: tools/delete_pubspec_overrides.dart [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import 'dart:io';
```
Comment (dcharkes) [inline tools/delete_pubspec_overrides.dart:1]: I guess we have 0 package dependencies in this script, so no pubspec is needed? it feels a bit weird though.

Does it mean this script always has to be the latest language version for example?

Comment (mosuem) [reply tools/delete_pubspec_overrides.dart:1]: I never use pubspecs on scripts :D It just means no restrictions on the SDK used, right?

Comment (dcharkes) [reply tools/delete_pubspec_overrides.dart:1]: Which could mean it stops working on Dart 4.0 ?

Fine with me either way. 4.0 is far out =P

**Review by dcharkes — APPROVED** [review]

**Pull request merged**